android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -625,6 +625,27 @@ public static String maybeHexStrToUpperCase(String value) {
             return value;
     }
 
+    public static long toDecimalLamport(String amount, int decimals) {
+        try {
+            amount = removeHexPrefix(amount);
+            BigDecimal value = new BigDecimal(amount);
+
+            return value.multiply(new BigDecimal(getDecimalsDepNumber(decimals)))
+                    .round(MathContext.UNLIMITED)
```

Comment:
do not round, with round you can have a situation that it round to a nearest higher number and when we pick that 100% amount to do something it will give an error that not enough balance, just strip as we do here https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java#L387

Reply:
fixed https://github.com/brave/brave-core/pull/14397/commits/c5dc6927b21d3bce27d2c348ad2d9cc88925d0ac